ElementX/Sources/Services/NotificationSettings/NotificationSettingsProxy.swift

```diff
@@ -111,10 +116,22 @@ final class NotificationSettingsProxy: NotificationSettingsProxyProtocol {
         defer { backgroundTask?.stop() }
 
         try await notificationSettings.setCallEnabled(enabled: enabled)
+        await waitForSettingsDidChange()
     }
     
     // MARK: - Private
     
+    func waitForSettingsDidChange(timeout duration: Duration = .seconds(2.0)) async {
```

Comment:
```suggestion
    func updatedSettings(timeout duration: Duration = .seconds(2.0)) async {
```

So that when we call it fells more natural:
```swift
await updatedSettings(...)
```

Reply:
👍